### PR TITLE
Add File and Dir empty? methods

### LINKED ISSUE #2951: Dir.empty? and File.empty?
The upcoming Ruby 2.4 brings `Dir.empty?` and `File.empty?` methods for finding out if a dir or a file is empty. See https://github.com/ruby/ruby/blob/trunk/NEWS

I think these methods are quite useful for some use cases and they could be handy as being a part of standard library.

### PR DESCRIPTION
**Pull request by dylandrop**
Adds `empty?` to the `File` and `Dir` classes. Closes #2951.

This follows the proposed behavior of [this comment](https://github.com/crystal-lang/crystal/issues/2951#issuecomment-230740384). This varies from Ruby in that an error is raised if the file / dir is not found.

In terms of implementation -- the File version is [pretty much identical](https://github.com/ruby/ruby/blob/trunk/file.c#L1749) to Ruby's. Interestingly, their implementation for Dir seems to have an optimization that leverages [getattrlist](https://github.com/ruby/ruby/blob/trunk/dir.c#L2670). I didn't know if this was super valuable, but it's an optimization that could be added in a later PR.

### CODE REVIEW

File: src/file.cr [hunk 1]
```diff
@@ -119,6 +119,23 @@ class File < IO::FileDescriptor
...
+  # File.write("foo", "")
+  # File.empty?("foo") # => true
+  # File.write("foo", "foo")
+  # File.empty?("foo") # => false
+  # ```
+  def self.empty?(path) : Bool
+    begin
+      stat(path).size == 0
+    rescue Errno
+      raise Errno.new("Error determining size of '#{path}'")
```
Comment (dylandrop) [inline src/file.cr]: This captures and re-raises the error, with a different message. I don't know if there's a more elegant / obvious way to do this.

**Comment by asterite** [comment]
@dylandrop I believe this is the correct way to implement these methods, the semantic you mention is pretty good :-)

I just made a tiny comment, after that I'll merge this.

File: src/dir.cr [hunk 2]
```diff
@@ -160,6 +160,24 @@ class Dir
...
+  # Dir.mkdir("bar")
+  # Dir.empty?("bar") # => true
+  # File.write("bar/a_file", "The content")
+  # Dir.empty?("bar") # => false
+  # ```
+  def self.empty?(path) : Bool
+    raise Errno.new("Error determining size of '#{path}'") unless exists?(path)
+
+    foreach(path) do |f|
+      return false unless %(. ..).includes?(f)
```
Comment (dylandrop) [inline src/dir.cr]: 😱 You're right -- originally I meant to do `%w`, but the tuple way is better. Updating.

**Comment by dylandrop** [comment]
@asterite Done.

**Comment by asterite** [comment]
@dylandrop Thank you for this! 💚

**Pull request merged**